pkg/recipes/terraform/execute.go

```diff
@@ -200,18 +201,15 @@ func (e *executor) GetRecipeMetadata(ctx context.Context, options Options) (map[
 	}, nil
 }
 
-// setEnvironmentVariables sets environment variables for the Terraform process by reading values from the environment configuration.
+// setEnvironmentVariables sets environment variables by reading values from the environment configuration.
 // Terraform process will use environment variables as input for the recipe deployment.
-func (e executor) setEnvironmentVariables(ctx context.Context, tf *tfexec.Terraform, envConfig *recipes.Configuration) error {
-	if envConfig != nil && envConfig.RecipeConfig.Env.AdditionalProperties != nil {
-		envVars := map[string]string{}
+func (e executor) setEnvironmentVariables(ctx context.Context, envConfig *recipes.Configuration) error {
```

Comment:
Missed this earlier, this function is only working recipe config, do we need entire env config for anything?

```suggestion
func (e executor) setEnvironmentVariables(ctx context.Context, recipeConfig *datamodel.RecipeConfigProperties) error {
```

Reply:
updated